Rector errors with anon class

<!-- First, thank you for reporting a bug. That takes time and we appreciate that! -->

| Subject        | Details                                                      |
| :------------- | :----------------------------------------------------------- |
| PHP version    | e.g. PHP 7.3                            |
| Full Command   | e.g. vendor/bin/rector process Foo.php  | 

## Current Behaviour 

```
 [ERROR] Could not process "Foo.php" file, due to:
         "Call to a member function getMethod() on null".

```

## Minimal PHP Code Causing Issue

```yaml
# rector.yaml
services:
    Rector\TypeDeclaration\Rector\Property\PropertyTypeDeclarationRector: ~
```

```php
class Foo
{
    private function makeClass(int $bar = 1)
    {
        $class =  new class() {
            public $bar;
        };
        $class->bar = $bar;
    }
}

```

## Expected Behaviour

no error, and:

```php
class Foo
{
    private function makeClass(int $bar = 1)
    {
        $class =  new class() {
            /** @var int */
            public $bar;
        };
        $class->bar = $bar;
    }
}

```

--- a/packages/TypeDeclaration/src/TypeInferer/PropertyTypeInferer/AllAssignNodePropertyTypeInferer.php
+++ b/packages/TypeDeclaration/src/TypeInferer/PropertyTypeInferer/AllAssignNodePropertyTypeInferer.php
@@ -4,6 +4,7 @@ namespace Rector\TypeDeclaration\TypeInferer\PropertyTypeInferer;
 
 use PhpParser\Node\Stmt\ClassLike;
 use PhpParser\Node\Stmt\Property;
+use PHPStan\Type\MixedType;
 use PHPStan\Type\Type;
 use Rector\NodeTypeResolver\Node\AttributeKey;
 use Rector\TypeDeclaration\Contract\TypeInferer\PropertyTypeInfererInterface;
@@ -24,8 +25,12 @@ final class AllAssignNodePropertyTypeInferer extends AbstractTypeInferer impleme
 
     public function inferProperty(Property $property): Type
     {
-        /** @var ClassLike $class */
+        /** @var ClassLike|null $class */
         $class = $property->getAttribute(AttributeKey::CLASS_NODE);
+        if ($class === null) {
+            // anonymous class
+            return new MixedType();
+        }
 
         $propertyName = $this->nameResolver->getName($property);
 

--- a/packages/TypeDeclaration/src/TypeInferer/PropertyTypeInferer/ConstructorPropertyTypeInferer.php
+++ b/packages/TypeDeclaration/src/TypeInferer/PropertyTypeInferer/ConstructorPropertyTypeInferer.php
@@ -28,8 +28,12 @@ final class ConstructorPropertyTypeInferer extends AbstractTypeInferer implement
 {
     public function inferProperty(Property $property): Type
     {
-        /** @var Class_ $class */
+        /** @var Class_|null $class */
         $class = $property->getAttribute(AttributeKey::CLASS_NODE);
+        if ($class === null) {
+            // anonymous class
+            return new MixedType();
+        }
 
         $classMethod = $class->getMethod('__construct');
         if ($classMethod === null) {

--- a/packages/TypeDeclaration/src/TypeInferer/PropertyTypeInferer/GetterPropertyTypeInferer.php
+++ b/packages/TypeDeclaration/src/TypeInferer/PropertyTypeInferer/GetterPropertyTypeInferer.php
@@ -45,8 +45,12 @@ final class GetterPropertyTypeInferer extends AbstractTypeInferer implements Pro
 
     public function inferProperty(Property $property): Type
     {
-        /** @var Class_ $class */
+        /** @var Class_|null $class */
         $class = $property->getAttribute(AttributeKey::CLASS_NODE);
+        if ($class === null) {
+            // anonymous class
+            return new MixedType();
+        }
 
         /** @var string $propertyName */
         $propertyName = $this->nameResolver->getName($property);

--- a/packages/TypeDeclaration/src/TypeInferer/PropertyTypeInferer/GetterTypeDeclarationPropertyTypeInferer.php
+++ b/packages/TypeDeclaration/src/TypeInferer/PropertyTypeInferer/GetterTypeDeclarationPropertyTypeInferer.php
@@ -29,8 +29,12 @@ final class GetterTypeDeclarationPropertyTypeInferer extends AbstractTypeInferer
 
     public function inferProperty(Property $property): Type
     {
-        /** @var Class_ $class */
+        /** @var Class_|null $class */
         $class = $property->getAttribute(AttributeKey::CLASS_NODE);
+        if ($class === null) {
+            // anonymous class
+            return new MixedType();
+        }
 
         /** @var string $propertyName */
         $propertyName = $this->nameResolver->getName($property);

--- a/packages/TypeDeclaration/src/TypeInferer/PropertyTypeInferer/SingleMethodAssignedNodePropertyTypeInferer.php
+++ b/packages/TypeDeclaration/src/TypeInferer/PropertyTypeInferer/SingleMethodAssignedNodePropertyTypeInferer.php
@@ -19,8 +19,12 @@ final class SingleMethodAssignedNodePropertyTypeInferer extends AbstractTypeInfe
 {
     public function inferProperty(Property $property): Type
     {
-        /** @var Class_ $class */
+        /** @var Class_|null $class */
         $class = $property->getAttribute(AttributeKey::CLASS_NODE);
+        if ($class === null) {
+            // anonymous class
+            return new MixedType();
+        }
 
         $classMethod = $class->getMethod('__construct');
         if ($classMethod === null) {

--- a/packages/TypeDeclaration/tests/Rector/Property/PropertyTypeDeclarationRector/Fixture/skip_anonymous_class.php.inc
+++ b/packages/TypeDeclaration/tests/Rector/Property/PropertyTypeDeclarationRector/Fixture/skip_anonymous_class.php.inc
@@ -1,0 +1,15 @@
+<?php
+
+namespace Rector\TypeDeclaration\Tests\Rector\FunctionLike\PropertyTypeDeclarationRector\Fixture;
+
+class SkipAnonymousClass
+{
+    public function makeClass(int $bar = 1)
+    {
+        $class = new class() {
+            public $bar;
+        };
+
+        $class->bar = $bar;
+    }
+}

--- a/packages/TypeDeclaration/tests/Rector/Property/PropertyTypeDeclarationRector/PropertyTypeDeclarationRectorTest.php
+++ b/packages/TypeDeclaration/tests/Rector/Property/PropertyTypeDeclarationRector/PropertyTypeDeclarationRectorTest.php
@@ -35,6 +35,7 @@ final class PropertyTypeDeclarationRectorTest extends AbstractRectorTestCase
         yield [__DIR__ . '/Fixture/doctrine_relation_target_entity_same_namespace.php.inc'];
         yield [__DIR__ . '/Fixture/setter_type.php.inc'];
         yield [__DIR__ . '/Fixture/skip_multi_vars.php.inc'];
+        yield [__DIR__ . '/Fixture/skip_anonymous_class.php.inc'];
     }
 
     protected function getRectorClass(): string

